### PR TITLE
[FIX][11.0] website_snippet_preset travis warning

### PR DESCRIPTION
**Pull request by tbaden**
issue https://github.com/OCA/website/issues/603 : 
fix of:
<details><summary>logs</summary>
```

************* Module website_snippet_preset
website_snippet_preset/static/src/js/website_snippet_preset.js:4: [W7903(javascript-lint), ] Use the function form of 'use strict'. [Error/strict]
website_snippet_preset/static/src/js/website_snippet_preset.js:4: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:13: [W7903(javascript-lint), ] Line 13 exceeds the maximum line length of 80. [Error/max-len]
website_snippet_preset/static/src/js/website_snippet_preset.js:16: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:18: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:22: [W7903(javascript-lint), ] ["name"] is better written in dot notation. [Error/dot-notation]
website_snippet_preset/static/src/js/website_snippet_preset.js:23: [W7903(javascript-lint), ] Expected space(s) after "if". [Error/keyword-spacing]
website_snippet_preset/static/src/js/website_snippet_preset.js:23: [W7903(javascript-lint), ] Expected { after 'if' condition. [Error/curly]
website_snippet_preset/static/src/js/website_snippet_preset.js:24: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:29: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:34: [W7903(javascript-lint), ] 'EnterNameDialog' was used before it was defined. [Error/no-use-before-define]
website_snippet_preset/static/src/js/website_snippet_preset.js:35: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:36: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:37: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:41: [W7903(javascript-lint), ] Expected space(s) after "if". [Error/keyword-spacing]
website_snippet_preset/static/src/js/website_snippet_preset.js:42: [W7903(javascript-lint), ] 'OverwriteDialog' was used before it was defined. [Error/no-use-before-define]
website_snippet_preset/static/src/js/website_snippet_preset.js:50: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:51: [W7903(javascript-lint), ] 'self' is assigned a value but never used. [Error/no-unused-vars]
website_snippet_preset/static/src/js/website_snippet_preset.js:56: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:57: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:58: [W7903(javascript-lint), ] Line 58 exceeds the maximum line length of 80. [Error/max-len]
website_snippet_preset/static/src/js/website_snippet_preset.js:58: [W7903(javascript-lint), ] Expected '===' and instead saw '=='. [Error/eqeqeq]
website_snippet_preset/static/src/js/website_snippet_preset.js:58: [W7903(javascript-lint), ] ["id"] is better written in dot notation. [Error/dot-notation]
website_snippet_preset/static/src/js/website_snippet_preset.js:62: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:67: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:68: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:69: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:74: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:77: [W7903(javascript-lint), ] Unexpected negated condition. [Error/no-negated-condition]
website_snippet_preset/static/src/js/website_snippet_preset.js:77: [W7903(javascript-lint), ] Expected { after 'if' condition. [Error/curly]
website_snippet_preset/static/src/js/website_snippet_preset.js:81: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:83: [W7903(javascript-lint), ] Expected { after 'else'. [Error/curly]
website_snippet_preset/static/src/js/website_snippet_preset.js:87: [W7903(javascript-lint), ] A space is required after ','. [Error/comma-spacing]
website_snippet_preset/static/src/js/website_snippet_preset.js:87: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:90: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:91: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:97: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:102: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:105: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:107: [W7903(javascript-lint), ] Line 107 exceeds the maximum line length of 80. [Error/max-len]
website_snippet_preset/static/src/js/website_snippet_preset.js:109: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:110: [W7903(javascript-lint), ] ["arch"] is better written in dot notation. [Error/dot-notation]
website_snippet_preset/static/src/js/website_snippet_preset.js:112: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:113: [W7903(javascript-lint), ] ["id"] is better written in dot notation. [Error/dot-notation]
website_snippet_preset/static/src/js/website_snippet_preset.js:119: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:125: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:127: [W7903(javascript-lint), ] Unexpected console statement. [Error/no-console]
website_snippet_preset/static/src/js/website_snippet_preset.js:132: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:136: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:137: [W7903(javascript-lint), ] 'self' is assigned a value but never used. [Error/no-unused-vars]
website_snippet_preset/static/src/js/website_snippet_preset.js:141: [W7903(javascript-lint), ] Line 141 exceeds the maximum line length of 80. [Error/max-len]
website_snippet_preset/static/src/js/website_snippet_preset.js:143: [W7903(javascript-lint), ] Line 143 exceeds the maximum line length of 80. [Error/max-len]
website_snippet_preset/static/src/js/website_snippet_preset.js:144: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:145: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:149: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:153: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:156: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:160: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:161: [W7903(javascript-lint), ] 'self' is assigned a value but never used. [Error/no-unused-vars]
website_snippet_preset/static/src/js/website_snippet_preset.js:164: [W7903(javascript-lint), ] Line 164 exceeds the maximum line length of 80. [Error/max-len]
website_snippet_preset/static/src/js/website_snippet_preset.js:165: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:168: [W7903(javascript-lint), ] Line 168 exceeds the maximum line length of 80. [Error/max-len]
website_snippet_preset/static/src/js/website_snippet_preset.js:168: [W7903(javascript-lint), ] There should be no space before '}'. [Error/object-curly-spacing]
website_snippet_preset/static/src/js/website_snippet_preset.js:169: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:170: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:174: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
website_snippet_preset/static/src/js/website_snippet_preset.js:176: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:181: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
website_snippet_preset/static/src/js/website_snippet_preset.js:182: [W7903(javascript-lint), ] Missing semicolon. [Error/semi]
```
</details>

### CODE REVIEW

**Review by yajo — CHANGES_REQUESTED** [review]
Still some failures: https://travis-ci.org/OCA/website/jobs/530017188#L923-L935

**Comment by tbaden** [comment]
still WIP, will check this later today

**Comment by tbaden** [comment]
@Yajo done

File: website_snippet_preset/static/src/js/website_snippet_preset.js [hunk 1]
```diff
@@ -1,44 +1,112 @@
...
-odoo.define('website_snippet_preset', function(require) {
+odoo.define('website_snippet_preset', function (require) {
+    'use strict';
     var snippet_editor = require('web_editor.snippet.editor');
     var core = require('web.core');
     var Dialog = require('web.Dialog');
     var qweb = core.qweb;
     var _t = core._t;
 
+    var OverwriteDialog = Dialog.extend({
```
Comment (yajo) [inline website_snippet_preset/static/src/js/website_snippet_preset.js:12]: Please respect the order in which things were declared.

**Review by yajo — CHANGES_REQUESTED** [review]

File: website_snippet_preset/static/src/js/website_snippet_preset.js (hunk 1 above)
Comment (tbaden) [reply website_snippet_preset/static/src/js/website_snippet_preset.js:12]: In this case you get a warning. Use before declaration.

Comment (yajo) [reply website_snippet_preset/static/src/js/website_snippet_preset.js:12]: Ah ok

**Comment by tbaden** [comment]
@Yajo anything else to do here?

**Review by yajo — APPROVED** [review]

**Pull request merged**